plugin-server/src/worker/ingestion/event-pipeline/runner.ts

```diff
@@ -8,60 +8,30 @@ import { timeoutGuard } from '../../../utils/db/utils'
 import { status } from '../../../utils/status'
 import { LazyPersonContainer } from '../lazy-person-container'
 import { generateEventDeadLetterQueueMessage } from '../utils'
-import { populateTeamDataStep } from './1-populateTeamDataStep'
-import { emitToBufferStep } from './2-emitToBufferStep'
-import { pluginsProcessEventStep } from './3-pluginsProcessEventStep'
-import { processPersonsStep } from './4-processPersonsStep'
-import { prepareEventStep } from './5-prepareEventStep'
-import { createEventStep } from './6-createEventStep'
-import { runAsyncHandlersStep } from './7-runAsyncHandlersStep'
-
-export type StepParameters<T extends (...args: any[]) => any> = T extends (
-    runner: EventPipelineRunner,
-    ...args: infer P
-) => any
-    ? P
-    : never
-
-const EVENT_PIPELINE_STEPS = {
-    populateTeamDataStep,
-    emitToBufferStep,
-    pluginsProcessEventStep,
-    processPersonsStep,
-    prepareEventStep,
-    createEventStep,
-    runAsyncHandlersStep,
-}
-
-export type EventPipelineStepsType = typeof EVENT_PIPELINE_STEPS
-export type StepType = keyof EventPipelineStepsType
-export type NextStep<Step extends StepType> = [StepType, StepParameters<EventPipelineStepsType[Step]>]
-
-export type StepResult =
-    | null
-    | NextStep<'populateTeamDataStep'>
-    | NextStep<'emitToBufferStep'>
-    | NextStep<'pluginsProcessEventStep'>
-    | NextStep<'processPersonsStep'>
-    | NextStep<'prepareEventStep'>
-    | NextStep<'createEventStep'>
-    | NextStep<'runAsyncHandlersStep'>
+import { createEventStep } from './createEventStep'
+import { emitToBufferStep } from './emitToBufferStep'
+import { pluginsProcessEventStep } from './pluginsProcessEventStep'
+import { populateTeamDataStep } from './populateTeamDataStep'
+import { prepareEventStep } from './prepareEventStep'
+import { processPersonsStep } from './processPersonsStep'
+import { runAsyncHandlersStep } from './runAsyncHandlersStep'
 
 // Only used in tests
```

Comment:
hmm... this comment makes me think that maybe we can nuke this (in a follow-up PR) and the code would be shorter easier to read & that doesn't necessarily convince me that tests do the right thing anyway. Then we could also remove the `registerLastStep` and instead just `statsd` & `return` when we exit early (assuming we agree that's more readable that way)

Reply:
Yes definitely. I left this in as the tests heavily depend on this and I didn't want to refactor those yet. Ideally we're ensure there are tests less coupled to the implementation of the `EventPipelineRunner` internal details and more to it's side-effects.

Reply:
Have added comment to remove.